frontend/pages/admin/TeamManagementPage/TeamManagementPage.tsx

```diff
@@ -93,21 +104,21 @@ const TeamManagementPage = (): JSX.Element => {
           dispatch(
             renderFlash("success", `Successfully created ${formData.name}.`)
           );
+          setBackendValidators({});
           toggleCreateTeamModal();
+          refetchTeams();
         })
-        .catch((createError: any) => {
+        .catch((createError: { data: IApiError }) => {
           if (createError.data.errors[0].reason.includes("Duplicate")) {
-            dispatch(
-              renderFlash("error", "A team with this name already exists.")
-            );
+            setBackendValidators({
+              name: "A team with this name already exists",
+            });
```

Comment:
I see here that the string parsing in the response object is already in place. This would be a good time to switch to checking response status codes for all of these instances. Status code checks are considered best practice vs. response object parsing for the reasons mentioned above.